### ISSUE TITLE
Add labels using baseline classifier for ranking sources

### ISSUE THREAD

**Issue by acmiyaguchi** [dsgt-birdclef/birdclef-2023#92]
We want to do our first semi-supervised labeling on our dataset. The goal here to rank each of the sound separated audio tracks by the frequency of the predicted label from the baseline model. We'll do this by predicting the class for each embedding in the consolidated_v3 dataset. For each row, we'll want to add the prediction from our model.

The model can be found under `data/models/baseline/logistic_full.pkl`. See https://www.kaggle.com/code/acmiyaguchi/birdnet-transfer-learning-inference for usage. You should be able to unpickle this and have a fully trained logistic regression model from sklearn.

You should read in the consolidated dataset, and add a new column that has the predicted column (and confidence). You should write the prediction using a pyspark udf. Then write out the dataframe.

```python
from functools import partial

F.udf("string")
def predict(model, vector):
   return model.predict(vector)

# df is the consolidated_v3 dataset
# model is the unpickled model
res = df.withColumn("predicted", partial(predict, model)).select("filename", "start_time", "predicted")

# write out res to a parquet file, using 1 or 2 partitions. Use the processed/birdnet-2023 folder and make a new dataset under there.
```

We can then use the predictions to rank sources by the one that is most likely to contain the class that we're interested in.

**Comment by acmiyaguchi** [comment]
- https://docs.databricks.com/udf/python.html
- https://spark.apache.org/docs/3.1.3/api/python/reference/api/pyspark.sql.functions.udf.html
